### PR TITLE
IndigoCheck2 and IndigoCheckObject2 refactored

### PR DESCRIPTION
**Pull request by even1024**
Old checks removed from indigo

### CODE REVIEW

File: api/c/indigo/src/indigo_misc.cpp [hunk 1]
```diff
@@ -1419,118 +1417,32 @@ CEXPORT const char* indigoCheckObj2(int item, const char* check_flags)
...
+        }
+        catch (Exception& e)
+        {
+            out.printf("{\"LOAD\":{\"message\":\"Error at loading structure. %s\"}}", e.message());
+        }
+
+        out.writeChar(0);
+        return tmp.string.ptr();
+    }
+    INDIGO_END(0);
```
Comment (mkviatkovskii) [inline api/c/indigo/src/indigo_misc.cpp:1443]: Better use `nullptr` here.

File: api/c/indigo/src/indigo_structure_checker.h [hunk 2]
```diff
@@ -43,8 +43,8 @@ namespace indigo
         CheckResult check(const IndigoObject& item, const std::vector<CheckTypeCode>& check_types = std::vector<CheckTypeCode>(),
                           const std::vector<int>& selected_atoms = std::vector<int>(), const std::vector<int>& selected_bonds = std::vector<int>());
 
-        std::string toJson(const StructureChecker2::CheckResult& res);
-        std::string toJson2(const StructureChecker2::CheckResult& res);
+        std::string toJson(const StructureChecker::CheckResult& res);
+        std::string toJson2(const StructureChecker::CheckResult& res);
```
Comment (mkviatkovskii) [inline api/c/indigo/src/indigo_structure_checker.h:47]: Do we still need both `toJson` and `toJson2` methods?

File: api/c/indigo/src/indigo_misc.cpp (hunk 1 above)
Comment (even1024) [reply api/c/indigo/src/indigo_misc.cpp:1443]: This is not nullptr. Here we return pointer to the null-terminated string. 0 here - means 0x0 character.

File: api/c/indigo/src/indigo_structure_checker.h (hunk 2 above)
Comment (even1024) [reply api/c/indigo/src/indigo_structure_checker.h]: fixed

File: api/c/indigo/src/indigo_misc.cpp (hunk 1 above)
Comment (SunFellow) [reply api/c/indigo/src/indigo_misc.cpp:1443]: @even1024, you are mistaken. 0 is the value returned by the function. Function return type is `const char *`. This means it returns null-pointer. 
If you want to return empty string, type: `INDIGO_END("");`

**Pull request merged**